app/components/collections/show/deposits_component.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Collections
+  module Show
+    # Component for rendering a list of deposits for the collection show page.
+    class DepositsComponent < ApplicationComponent
+      def initialize(presenter:)
+        @presenter = presenter
+        super()
+      end
+      attr_reader :presenter
+
+      delegate :collection, :work_statuses, to: :presenter
+
+      def collection_deposits
+        @collection_deposits ||= collection.works.map do |work|
+          {
+            id: work.druid || work.id,
+            values: values_for(work)
+          }
+        end
+      end
+
+      def values_for(work) # rubocop:disable Metrics/AbcSize
+        content = content_for(work)
+        [
+          helpers.link_to(work.title, link_for(work)),
+          work.user.name,
+          status_for(work),
+          work.object_updated_at ? helpers.l(work.object_updated_at, format: '%b %d, %Y') : nil,
+          content.nil? ? 'N/A (still depositing)' : content.content_files.count,
+          content.nil? ? 'N/A (still depositing)' : helpers.number_to_human_size(content.content_files.sum(:size)),
+          persistent_link_for(work)
+        ]
+      end
+
+      def content_for(work)
+        return if work.druid.blank?
+
+        Contents::Builder.call(
```

Comment:
1. We shouldn't be doing `Sdr::Repository.find` in a component.
2. We can get the file count and sizes from the cocina without calling `Contents::Builder`.
3. I'd prefer not to do any of this, as it is going to be very expensive. Let me ask.

Reply:
Sounds good, @justinlittman